signing/src/main/java/tech/pegasys/web3signer/signing/secp256k1/aws/CachedAwsKmsClientFactory.java

```diff
@@ -45,22 +49,31 @@ public CachedAwsKmsClientFactory(final long cacheSize) {
                 new CacheLoader<>() {
                   @Override
                   public AwsKmsClient load(final AwsKmsClientKey key) {
+                    final AwsCredentialsProvider awsCredentialsProvider =
+                        AwsCredentialsProviderFactory.createAwsCredentialsProvider(
+                            key.getAwsAuthenticationMode(), key.getAwsCredentials());
+
                     final KmsClientBuilder kmsClientBuilder = KmsClient.builder();
                     key.getEndpointOverride().ifPresent(kmsClientBuilder::endpointOverride);
+                    final String region =
+                        key.getAwsAuthenticationMode() == AwsAuthenticationMode.SPECIFIED
+                            ? key.getRegion()
+                            : System.getenv("AWS_REGION");
```

Comment:
Intrigued to see if RegionProvider can be used here as @usmansaleem mentioned because I'm concerned ENVIRONMENT mode isn't guaranteed to have AWS_REGION envvar set?

Reply:
Changed to use the AwsRegionProvider